tests/server/models/relations-test.js

```diff
@@ -7,259 +7,216 @@ var Order = require('../../../server/db/models/order');
 var Category = require('../../../server/db/models/category');
 var Review = require('../../../server/db/models/review');
 var db = require('../../../server/db/_db');
+var OrderProducts=require('../../../server/db/models/order-products');
 var Promise=require('bluebird');
 
-describe('Category',function(){
+describe('Relations',function(){ 
+
+  var categoriesArr;
+  var productsArr;
+  var reviewsArr;
+  var ordersArr;
+  var user;
+
   beforeEach(function () {
     return db.sync({force: true});
   });
-  it('can be associated with multiple products',function(){
-    var productsArr;
-    var category1=Category.create({
-             name: 'Leisure'});
-    var product1=Product.create({ name: 'Unicorn robot', description: 'Most beautiful creature', inventory: 10, currentPrice: 100.00});
-    var product2=Product.create({ name: 'Unicorn puppy', description: '2nd most beautiful creature', inventory: 10, currentPrice: 100.00});
+  
+  beforeEach(function (done) {
+      var category1=Category.create({name: 'Leisure'});
+      var category2=Category.create({name: 'Evil'});
+      var product1=Product.create({ name: 'Unicorn robot', description: 'Most beautiful creature', inventory: 10, currentPrice: 100.00});
+      var product2=Product.create({ name: 'Unicorn puppy', description: '2nd most beautiful creature', inventory: 10, currentPrice: 100.00});
+      var review1=Review.create({
+                 stars: 5,
+                  content: 'the best damn unicorn i ever rode'});
+      var review2=Review.create({
+                 stars: 1,
+                  content: 'the worst evilest robot i ever saw'});
+      var user1=User.create({email: 'bob@bob.com', name: 'bob', password:'ohman',isAdmin:'false'});
+      var order1=Order.create({shippingAddress: 'the cupboard'});
+      var order2=Order.create({shippingAddress: 'faraway'});
 
-    return Promise.all([category1,product1,product2])
-    .spread(function(category,product1,product2){
-      productsArr=[product1,product2]
-      return category.addProducts(productsArr);
-    })
-    .then(function(){
-      return Category.findOne({
-            where: { name: 'Leisure' },
-            include: { model: Product }
-          })
-    })
-    .then(function(category){
-      expect(category.products[0].name).to.equal('Unicorn robot');
-      expect(category.products[1].name).to.equal('Unicorn puppy');
-    })
+      Promise.all([category1,category2,product1,product2,review1,review2,user1,order1,order2])
+      .spread(function(category1,category2,product1,product2,review1,review2,user1,order1,order2){
```

Comment:
Always be careful. Try and name function parameters differently from variable names so you don't encounter accidental overlap